students/jeff_shabani/session07/html_render.py

```diff
@@ -8,21 +8,205 @@
 # This is the framework for the base class
 class Element(object):
 
-    def __init__(self, content=None):
-        self.content = str(content)
+    indent = " "*2
+
+    def __init__(self, content=None, tag=None, **attrs):
+        self.attrs = attrs
+        if tag:
+            self.tag = tag
+        else:
+            self.tag = ''
+        if content:
+            self.content = content
+        else:
+            self.content = ''
 
     def append(self, new_content):
         result = list()
         result.append(new_content)
         for i in result:
-            self.content +=i
+            self.content += f'\n{i}\n'
         return self.content
 
+    def render(self, file_name, cur_indent=''):
```

Comment:
For the render, the input parameter should not be called _file_name_, it's really a _StringIO_ object, acts more like a file handler. In any case, this means you can write directly to it by calling its _write()_ method, saving the contents to a file will occur in _run_html_render.py_.  It its simplest form, this method should look like this:

        for content_line in self.content:
            file_name.write(f'{content_line}\n')